File: in-js/leetcode/qno-1.js
function temp(nums, target) {
    let obj = {}

    for (let i = 0; i < nums.length; i++) {
        let diff = target - nums[i]

        if (diff in obj) {
            return [obj[diff], i]
        }
        obj[nums[i]] = i
    }
    return []
}

console.log(temp([2, 7, 11, 15], 9));

File: in-rust/leetcode/src/qno-1/qno-1.rs
use std::collections::HashMap;

fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
    let mut hm = HashMap::new();
    println!("hm are: {:?}", hm);
    for (i, &num) in nums.iter().enumerate() {
        if let Some(&j) = hm.get(&num) {
            // If the complement exists, return the indices
            return vec![j as i32, i as i32];
        }
        // Insert the complement of the current element (target - num) with its index
        hm.insert(target - num, i);
    }
    vec![] // Return an empty vector if no solution is found
}

fn main() {
    let nums = vec![2, 7, 11, 15];
    let target = 9;
    let indexes = two_sum(nums, target);
    println!("Indexes are: {:?}", indexes);
}